integration/oplog/basic_oplog_test.go

```diff
@@ -0,0 +1,207 @@
+				bson.D{{"_id", int64(2)}, {"foo2", "bar2"}},
+				bson.D{{"_id", int64(3)}, {"foo3", "bar3"}},
+			})
+			require.NoError(t, err)
+
+			var lastOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"ns", ns}}, opts).Decode(&lastOplogEntry)
+			require.NoError(t, err)
+
+			actual := integration.ConvertDocument(t, lastOplogEntry)
+			unsetUnusedOplogFields(actual)
+			actualKeys := actual.Keys()
+			assert.ElementsMatch(t, expectedKeys, actualKeys)
+
+			// The last oplog entry should be the last insert.
+			expected, err := types.NewDocument(
+				"op", "i",
+				"ns", ns,
+				"o", must.NotFail(types.NewDocument("_id", int64(3), "foo3", "bar3")),
+				"ts", must.NotFail(actual.Get("ts")).(types.Timestamp),
+				"v", expectedV,
+			)
+			require.NoError(t, err)
+			assert.Equal(t, expected, actual)
+		})
+	})
+
+	t.Run("Delete", func(tt *testing.T) {
+		tt.Run("SingleDocument", func(tt *testing.T) {
+			t := setup.FailsForFerretDB(tt, "https://github.com/FerretDB/FerretDB/issues/3556")
+
+			_, err := coll.DeleteOne(ctx, bson.D{{"_id", int64(1)}})
+			require.NoError(t, err)
+
+			var lastOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"ns", ns}}, opts).Decode(&lastOplogEntry)
+			require.NoError(t, err)
+
+			actual := integration.ConvertDocument(t, lastOplogEntry)
+			unsetUnusedOplogFields(actual)
+			actualKeys := actual.Keys()
+			assert.ElementsMatch(t, expectedKeys, actualKeys)
+
+			expected, err := types.NewDocument(
+				"op", "d",
+				"ns", ns,
+				"o", must.NotFail(types.NewDocument("_id", int64(1))),
+				"ts", must.NotFail(actual.Get("ts")).(types.Timestamp),
+				"v", expectedV,
+			)
+			require.NoError(t, err)
+			assert.Equal(t, expected, actual)
+
+			// If an entry to delete is not found, expect oplog not to be written.
+			_, err = coll.DeleteOne(ctx, bson.D{{"_id", "non-existent"}})
+			require.NoError(t, err)
+
+			var newOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"ns", ns}}, opts).Decode(&newOplogEntry)
+			require.NoError(t, err)
+			assert.Equal(t, lastOplogEntry, newOplogEntry)
+		})
+
+		tt.Run("MultipleDocuments", func(tt *testing.T) {
+			t := setup.FailsForFerretDB(tt, "https://github.com/FerretDB/FerretDB/issues/3556")
+
+			_, err := coll.DeleteMany(ctx, bson.D{{"_id", bson.D{{"$gte", int64(2)}}}})
+			require.NoError(t, err)
+
+			var lastOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"o.applyOps.ns", ns}}, opts).Decode(&lastOplogEntry)
+			require.NoError(t, err)
+
+			actual := integration.ConvertDocument(t, lastOplogEntry)
+			unsetUnusedOplogFields(actual)
+			actualKeys := actual.Keys()
+			assert.ElementsMatch(t, expectedKeys, actualKeys)
+
+			ui := must.NotFail(must.NotFail(must.NotFail(must.NotFail(actual.Get("o")).(*types.Document).Get("applyOps")).(*types.Array).Get(0)).(*types.Document).Get("ui")).(types.Binary) //nolint:lll // splitting won't make it more readable
```

Comment:
I would argue that splitting it right in the middle would make it more readable:
```
applyOps := …
ui := applyOps.Get(0). …
```